frescobaldi_app/vcs/gitjob.py

```diff
@@ -164,15 +164,14 @@ def _finished(self, exitcode, exitstatus):
         """
         Called when the process has completed.
         process results and forward the signal
-
-        If error occurrs during the execution process, signal 'errorOccurred' 
-        will be emitted. Signal 'finished' will be blocked. So this function 
-        only works when process executes successfully. 
-        """
-        # Only handles the result when process exit normaly. 
-        if exitstatus == QProcess.NormalExit:
-            self._handle_results()
-            self.finished.emit(self)
+        
+        exitcode == 0 : Process executes successfully. 
+                        Result will output to stdout().
+        exitcode == 1 : Process returns an error message.
+                        Error message will output to stderr() 
+        """ 
+        self._handle_results()
+        self.finished.emit(exitcode)
```

Comment:
(Oh.... I need to put my question here)
Should I define an enum instead of emitting an int in signal finished()?